src/default_start.rs

```diff
@@ -248,14 +268,11 @@ pub async fn start<Mapping: 'static + ServiceMapping>(
     let consensus_interval = current_consensus_status.consensus_interval;
     let status_agent = StatusAgent::new(current_consensus_status);
 
-    assert_eq!(verifier_list.len(), config.consensus.public_keys.len());
     let mut bls_pub_keys = HashMap::new();
-    for (validator, bls_pub_key) in verifier_list
-        .iter()
-        .zip(config.consensus.public_keys.iter())
-    {
-        let address = validator.address.as_bytes();
-        let hex_pubkey = hex::decode(bls_pub_key).map_err(MainError::FromHex)?;
+    for validator_extend in metadata.verifier_list.iter() {
+        let address = validator_extend.address.as_bytes();
+        let hex_pubkey =
```

Comment:
It only changes how bls_pub_key is read when starting. Should check that when bls_pub_key changes in metadata service, the consensus module gets the latest bls_pub_key. 
You can refer how consensus module gets the latest verifier list.